docs/aws/audit/redshiftmonitoring/rules/redshift_cluster_encrypted.mdx

```diff
@@ -23,6 +23,219 @@ HIPAA, GDPR, NIST, SOC2, HITRUST, PCIDSS, NISTCSF
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Sign in to the AWS Management Console.
+2. Navigate to the Redshift dashboard by selecting "Services" from the top menu, then selecting "Redshift" under the "Database" category.
+3. In the Redshift dashboard, select "Clusters" from the left-hand menu. This will display a list of all your Redshift clusters.
+4. For each cluster, check the "Encryption" column. If the value is "Disabled" or "Not enabled", then the Redshift cluster is not encrypted.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to install and configure AWS CLI on your local machine. Make sure you have the necessary permissions to access Redshift resources.
+
+2. Use the following AWS CLI command to list all the Redshift clusters:
+
+   ```
+   aws redshift describe-clusters --query 'Clusters[*].[ClusterIdentifier]' --output text
+   ```
+   This command will return the identifiers of all Redshift clusters.
+
+3. For each cluster identifier returned from the previous step, run the following command to get the encryption status:
+
+   ```
+   aws redshift describe-clusters --cluster-identifier <ClusterIdentifier> --query 'Clusters[*].[Encrypted]' --output text
+   ```
+   Replace `<ClusterIdentifier>` with the actual cluster identifier. This command will return whether the cluster is encrypted or not.
+
+4. If the output of the above command is 'false', then the Redshift cluster is not encrypted. If it's 'true', then the cluster is encrypted. Repeat steps 3 and 4 for each cluster identifier to check the encryption status of all Redshift clusters.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install and configure AWS SDK for Python (Boto3) on your local system. This will allow you to interact with AWS services including Redshift.
+
+```python
+pip install boto3
+aws configure
+```
+
+2. Import the necessary modules and create a Redshift client using your AWS credentials.
+
+```python
+import boto3
+
+client = boto3.client(
+    'redshift',
+    region_name='us-west-2',
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
+)
+```
+
+3. Use the `describe_clusters` method to retrieve information about all Redshift clusters in your account.
+
+```python
+clusters = client.describe_clusters()
+```
+
+4. Iterate over the clusters and check the `Encrypted` attribute. If it's set to `False`, the cluster is not encrypted.
+
+```python
+for cluster in clusters['Clusters']:
+    if not cluster['Encrypted']:
+        print(f"Cluster {cluster['ClusterIdentifier']} is not encrypted.")
+```
+
+This script will print the identifiers of all unencrypted Redshift clusters. If no output is produced, all clusters are encrypted.
+</Accordion>
+
+</AccordionGroup>
+</Tab>
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>